src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -310,9 +310,27 @@ SHAMapStoreImp::run()
             state_db_.setLastRotated(lastRotated);
         }
 
+        // Make sure we don't delete ledgers currently being
+        // imported into the ShardStore
+        bool const waitForImport = [this, lastRotated] {
+            if (auto shardStore = app_.getShardStore())
+            {
+                if (auto sequence = shardStore->getDatabaseImportSequence())
+                    return sequence <= lastRotated - 1;
+            }
+
+            return false;
+        }();
+
+        if (waitForImport)
+        {
+            JLOG(journal_.trace())
+                << "rotation would interfere with ShardStore import";
+        }
```

Comment:
Done. I just simplified the info log condition to `if (waitForImport)` since `waitForImport` is only true when `readyToRotate` is also true based on the new lambda.